src/app/api/volunteers/[volunteerId]/route.ts

```diff
@@ -0,0 +1,44 @@
+import { updateVolunteer } from '@/server/actions/Volunteer';
+import { zObjectId } from '@/types/dataModel/base';
+import {
+  zCreateVolunteerRequest,
+  zVolunteerResponse,
+} from '@/types/dataModel/volunteer';
+import { NextRequest, NextResponse } from 'next/server';
+
+// @route PUT /api/volunteers/[volunteerId] - updates an existing volunteer in the database
+export async function PUT(
+  request: NextRequest,
+  { params }: { params: { volunteerId: string } }
+) {
+  const validationResult = zObjectId.safeParse(params.volunteerId);
+  if (!validationResult.success) {
+    return NextResponse.json(
+      { message: 'Invalid Volunteer ID' },
+      { status: 400 }
+    );
+  }
+
+  const vId = validationResult.data;
+
+  // Get the updated volunteer data
+  const res = await request.json();
+  const volunteerRequest = zCreateVolunteerRequest.safeParse(res);
+
+  let updatedVolunteer;
+  if (volunteerRequest.success) {
+    updatedVolunteer = await updateVolunteer(vId, volunteerRequest.data);
+  } else {
+    return NextResponse.json(
```

Comment:
this will just be a 400 error i believe, this means that the `volunteerRequest` is not in the correct shape